website/blog/2022-06-04-2.7.0.md

```diff
@@ -0,0 +1,441 @@
+---
+author: "Sosuke Suzuki (@sosukesuzuki)"
+authorURL: "https://github.com/sosukesuzuki"
+title: "Prettier 2.7: new --cache CLI option and TypeScript 4.7 syntax!"
+---
+
+This release includes a new `--cache` CLI option. Enabling this option will use some attributes as cache keys and format files only if they have changed. This could dramatically improve CLI performance.
+
+We've also added support formatting for TypeScript 4.7 syntax!
+
+## Highlights
+
+### TypeScript
+
+#### Support TypeScript 4.7 ([#12896](https://github.com/prettier/prettier/pull/12896), [#12897](https://github.com/prettier/prettier/pull/12897), [#12898](https://github.com/prettier/prettier/pull/12898), [#12900](https://github.com/prettier/prettier/pull/12900), [#12921](https://github.com/prettier/prettier/pull/12921), [#12924](https://github.com/prettier/prettier/pull/12924), [#12959](https://github.com/prettier/prettier/pull/12959) by [@sosukesuzuki](https://github.com/sosukesuzuki))
+
+Support [TypeScript 4.7](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7) new features!
+
+##### [Instantiation Expressions](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#instantiation-expressions)
+
+```ts
+interface Box<T> {
+  value: T;
+}
+function makeBox<T>(value: T) {
+  return { value };
+}
+const makeHammerBox = makeBox<Hammer>;
+const makeWrenchBox = makeBox<Wrench>;
+```
+
+##### [Optional Variance](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#optional-variance-annotations-for-type-parameters)
+
+```ts
+interface Animal {
+  animalStuff: any;
+}
+interface Dog extends Animal {
+  dogStuff: any;
+}
+type Getter<out T> = () => T;
+type Setter<in T> = (value: T) => void;
+```
+
+##### [`extends` constraints for `infer`](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#extends-constraints-on-infer-type-variables)
+
+```ts
+type FirstString<T> = T extends [infer extends string, ...unknown[]]
```

Comment:
```suggestion
type FirstString<T> = T extends [infer S extends string, ...unknown[]]
```

Reply:
Thank you! [ee06ea9](https://github.com/prettier/prettier/pull/12966/commits/ee06ea9fc1464ded8a8e4bcbda022923743a08cf)